src/components/LocationSelector/LocationSelector.tsx

```diff
@@ -1,17 +1,82 @@
+    /** Regional setting for the map. By default Google uses US.
+     * For details see: https://developers.google.com/maps/documentation/javascript/localization#Region
+     */
+    region?: string;
+    /** other props to pass to the google loader. For details see: https://react-google-maps-api-docs.netlify.com/#loadscriptnext */
+    additionalGoogleProps?: object; // eslint-disable-line react/forbid-prop-types
+    /** stores an array of selected location objects */
+    selectedLocations: Location[];
+    /** defines if selector has an option to control the radius for a marker */
+    hasRadius?: boolean;
+    /** default radius value */
+    radiusDefaultValue?: number;
+    /** radius measurement unit */
+    radiusUnits: 'km' | 'mi';
+    /** radius label renderer e.g. radius => `+ ${radius} km` */
+    renderRadiusLabel: (string) => string;
+    /** min radius value of the slider component */
+    minRadius?: number;
+    /** max radius value of the slider component */
+    maxRadius?: number;
+    /** radius step value of the slider component */
+    radiusStep?: number;
+    /** defines if selector has a list of locations cards to render */
+    withoutLocationCards?: boolean;
+    /** country where search can take place */
+    country: string;
+    /** address to make initial map centering more specific */
+    initialMapAddress?: string;
+    /**
+     * type of locations that should be searched for.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletionRequest.types
+     */
+    placeTypes?: string[];
+    /** show country name in autocomplete suggestions */
+    showCountryInSuggestions?: boolean;
+    /** A title for the modal that will be used by screenreaders */
+    modalContentLabel: string;
+    /** placeholder for both main field and autocomplete field in modal */
+    inputPlaceholder: string;
+    /** placeholder for empty LocationAutocomplete list */
+    noSuggestionsPlaceholder: string;
+    /** function to be executed if error occurs while fetching suggestions */
+    onLocationAutocompleteError: () => void;
+    /** string to be displayed in FieldWrapper when the modal is closed, but locations are selected */
+    selectionPlaceholder?: string;
+    /** label for the Done button */
+    doneLabel: string;
+    /** label to be used on the clear all button */
+    clearLabel: string;
+    /** function called with location object as an argument when it is selected from the suggestions */
+    onAddLocation: (Location) => void;
+    /** function called with a location details as an argument to be changed */
+    onUpdateLocation: (Location) => void;
+    /** function with a locationId as an argument to be removed */
+    onRemoveLocation: (Location) => void;
+    /** callback function for the Clear button click */
+    onRemoveAllLocations: () => void;
+    /** callback function for closed modal */
+    onBlur: () => void;
```

Comment:
Do you think we need to mark that PR that has breaking changes? After increasing the OneUI version in JF can that cause some `typescript` errors?

Reply:
I've just converted `js` file to `tsx`. I guess we can do that but I'm not sure if it's really necessary. Current version of OneUI includes breaking changes what makes future updates of JF safe enough. Also api has not been changed and LocationSelector already has all of the required props included.

Reply:
Only mark breaking change if it was introduced with this PR and when it involves a particular change that hasn't occurred / was already marked as a breaking change before.